src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs

```diff
@@ -108,6 +108,12 @@ public NoteViewModel(WorkspaceViewModel workspaceViewModel, NoteModel model)
             ZIndex = ++StaticZIndex; // places the note on top of all nodes/notes
         }
 
+        public virtual void Dispose()
+        {
+            _model.PropertyChanged -= note_PropertyChanged;
+            DynamoSelection.Instance.Selection.CollectionChanged -= SelectionOnCollectionChanged;
```

Comment:
should there be dispose on viewmodelbase covering few dispose (like selection)?

Reply:
I will give that a shot - I think the base implementation may be empty (nothing shared between all viewModels, but thats okay, it will be more clear.

Reply:
Possibly the base implementation should be abstract to force all derived classes to properly dispose their contents?

Reply:
still the base class can handle few common things?